packages/plugin-typescript/src/lib/runner/runner.ts

```diff
@@ -0,0 +1,60 @@
+import type {
+  AuditOutput,
+  AuditOutputs,
+  AuditReport,
+  Issue,
+  RunnerFunction,
+} from '@code-pushup/models';
+import type { AuditSlug } from '../types.js';
+import {
+  type DiagnosticsOptions,
+  getTypeScriptDiagnostics,
+} from './ts-runner.js';
+import type { CodeRangeName } from './types.js';
+import { getIssueFromDiagnostic, tSCodeToAuditSlug } from './utils.js';
+
+export type RunnerOptions = DiagnosticsOptions & {
+  expectedAudits: { slug: AuditSlug }[];
+};
+
+export function createRunnerFunction(options: RunnerOptions): RunnerFunction {
+  const { tsconfig, expectedAudits } = options;
+  return async (): Promise<AuditOutputs> => {
+    const diagnostics = await getTypeScriptDiagnostics({ tsconfig });
+    const result: Record<
+      CodeRangeName,
+      Pick<AuditReport, 'slug' | 'details'>
+    > = diagnostics.reduce(
+      (acc, diag) => {
+        const slug = tSCodeToAuditSlug(diag.code);
+        const existingIssues: Issue[] =
+          (acc[slug] && acc[slug].details?.issues) || ([] as Issue[]);
```

Comment:
I see unnecessary casting :arrow_right: I complain :grin: 

The whole expression can be simplified, actually:

```suggestion
        const existingIssues: Issue[] = acc[slug]?.details?.issues ?? [];
```

Reply:
I think this change was overwritten by applying suggestion from https://github.com/code-pushup/cli/pull/932#discussion_r1952869368